jax_galsim/spergel.py

```diff
@@ -0,0 +1,470 @@
+        c6 = z2 * (16.0 + z2 - 8.0 * nu) * c3
+        return c1 * (c4 * c5 * c2 + jnp.power(4.0, nu) * (c6 + 32.0 * _gamma(nu)))
+
+    return jnp.select(
+        [nu == 0, nu == 1, nu == 2, nu == 3, nu == 4],
+        [f0(z), f1(z), f2(z), f3(z), f4(z)],
+        default=fnu(z, nu),
+    )
+
+
+@jax.jit
+def fz_nu(z, nu):
+    """z^nu K_nu[z], z > 0"""
+    return jnp.where(z <= 1.0e-10, fsmallz_nu(z, nu), jnp.power(z, nu) * _Knu(nu, z))
+
+
+@jax.jit
+def fsmallz_nup1(z, nu):
+    def fnu(z, nu):
+        """z^(nu+1) K_(nu+1)[z] z -> 0"""
+        z2 = z * z
+        z4 = z2 * z2
+        c1 = -jnp.power(2.0, -4.0 - nu)
+        c2 = _gamma(-2.0 - nu)
+        c3 = c1 * c2 * (8.0 + 4.0 * nu + z2) * jnp.power(z, 2.0 * (1.0 + nu))
+        c4 = jnp.power(2.0, nu)
+        c5 = _gammap1(nu)
+        c6 = c4 * c5 * (1.0 - 0.25 * z2 / nu + z4 * 0.03125 / (nu * (nu - 1.0)))
+        return c3 + c6
+
+    return jnp.select(
+        [nu == 0, nu == 1, nu == 2, nu == 3, nu == 4],
+        [f1(z), f2(z), f3(z), f4(z), f5(z)],
+        default=fnu(z, nu),
+    )
+
+
+@jax.jit
+def fz_nup1(z, nu):
+    """z^(nu+1) K_{nu+1}(z)"""
+    return jnp.where(
+        z <= 1.0e-10, fsmallz_nup1(z, nu), jnp.power(z, nu + 1.0) * _Knu(nu + 1.0, z)
+    )
+
+
+@jax.jit
+def fluxfractionFunc(z, nu, alpha):
+    """1 - z^(nu+1) K_{nu+1}(z) / (2^nu Gamma(nu+1)) - alpha"""
+    return 1.0 - fz_nup1(z, nu) / (jnp.power(2.0, nu) * _gammap1(nu)) - alpha
+
+
+@jax.jit
+def reducedfluxfractionFunc(z, nu, norm):
+    """(1 - z^(nu+1) K_{nu+1}(z) / (2^nu Gamma(nu+1)))/norm"""
+    return fluxfractionFunc(z, nu, alpha=0.0) / norm
+
+
+@jax.jit
+def calculateFluxRadius(alpha, nu, zmin=0.0, zmax=30.0):
+    """Return radius R enclosing flux fraction alpha  in unit of the scale radius r0
+
+    Method: Solve  F(R/r0=z)/Flux - alpha = 0 using bisection algorithm
+
+    F(R)/F =  int( 1/(2^nu Gamma(nu+1)) (r/r0)^(nu+1) K_nu(r/r0) dr/r0; r=0..R) = alpha
+    =>
+    z=R/r0 such that
+    1 - z^(nu+1) K_{nu+1}(z) / (2^nu Gamma(nu+1)) = alpha
+
+    Typical use cases:
+     o alpha = 1/2 => R = Half-Light-Radius,
+     o alpha = 1 - folding-thresold => R used for stepk computation
+
+     nu: the Spergel index
+
+     nb. it is supposed that nu is in [-0.85, 4.0] checked in the Spergel class init
+    """
+    return bisect_for_root(
+        partial(fluxfractionFunc, nu=nu, alpha=alpha), zmin, zmax, niter=75
+    )
+
+
+@_wraps(
+    _galsim.Spergel,
+    lax_description="""
+    The JAX version uses the following profile
+        .. math::
+
+            I(r) = flux \times \left(2\pi 2^\nu \Gamma(1+\nu) r_0^2\right)^{-1}
+              \times \left(\frac{r}{r_0}\right)^\nu K_\nu\left(\frac{r}{r_0}\right)
+
+    with the following Fourier expression
+        .. math::
+
+            \hat{I}(k) = flux / (1 + (k r_0)^2)^{1+\nu}
+
+    where :math:`r_0` is the ``scale_radius``, and :math: `\nu` mandatory to be in [-0.85,4.0]
+    """,
```

Comment:
I'm confused by this. Does this convention mean that the spergel profile in galsim vs jax_galsim with the same flux will produce different surface brightness values?

Reply:
Hi @beckermr , sorry if I have introduced  confusing comments. 
I have followed the same definition as Galsim code, so in principle all the physics paramters are the same. Now, I thought that the Galsim class comment is not so clear: 
1.  most importantly compared to the original Spergel's paper for instance, the parameter r0 in Galsim is the scale_factor and not the half-light-radius as Spergel notation. So it can make people confusing
2. the nu parameter in the JAX-Galsim code is not asserted to be in the [-0.86,4.0] interval (ie. no JAX way to do it till we find a way), so we need to specify this somewhere.
3. I prefer personally to write the complete expression and no let the user discovering it reading the code.

Reply:
I suggested a change in wording to resolve this.